website/docs/soft_delete.md

```diff
@@ -0,0 +1,80 @@
+---
+id: soft_delete
+title: Soft Delete Handler
+sidebar_label: Soft Delete Handler
+---
+
+## Soft Delete
+In `Core-Lib`, the classes `SoftDeleteMixin` and `SoftDeleteTokenMixin` handle soft deletes. These two classes can be imported 
+and passed to a `SQLAlchemy` Base class to create the appropriate soft delete fields.
+
+### SoftDeleteMixin
+`SoftDeleteMixin` creates the following columns in a table.
+- `created_at`:
+  - type: `DateTime`
+  - default: `datetime.utcnow`
+
+
+- `updated_at`:
+  - type: `DateTime`
+  - default: `datetime.utcnow`
+  - onupdate: `datetime.utcnow` immediately updates when the system changes the value of `deleted_at`.
+
+
+- `deleted_at`:
+  - type: `DateTime`
+  - default: `None`
+  - When a user wishes to execute a soft deletion on a column, this field must be updated by the value `datetime.utcnow`.
+
+```python
+class SoftDeleteMixin(object):
+    created_at = Column(DateTime, default=datetime.utcnow)
+    created_at._creation_order = 9998
+    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
+    updated_at._creation_order = 9998
+    deleted_at = Column(DateTime, default=None)
+    deleted_at._creation_order = 9998
+```
+
+
+### SoftDeleteTokenMixin
+`SoftDeleteTokenMixin` creates the following column in a table. Which can be used for Indexing if the user wants.

```

Comment:
forgot to mention that. 
When index is needed on deleted_at column. it's better to user on the delete_token for reason that index on a datetime column are very slow. 

This is the only reason for this class to exists,. so it's worth mentioning that